lib/config/folderconfiguration.go

```diff
@@ -226,6 +226,12 @@ func (f *FolderConfiguration) prepare(myID protocol.DeviceID, existingDevices ma
 	} else if f.MaxConcurrentWrites > maxConcurrentWritesLimit {
 		f.MaxConcurrentWrites = maxConcurrentWritesLimit
 	}
+
+	if f.Type == FolderTypeReceiveEncrypted {
+		f.IgnorePerms = true
+		f.Versioning.Type = ""
+		f.Versioning.Params = nil
+	}
```

Comment:
I see, it is in fact enforced by the backend.  Maybe we should have an API endpoint to receive the default folder config with "encrypted" as a parameter?

Reply:
Sounds reasonable,. Any kind of defaults aren't available though as of yet: https://github.com/syncthing/syncthing/pull/7131

Reply:
So this can go into a follow-up PR once #7131 is done.  Should we open an issue as reminder?

Reply:
In addition, does the "Watch for changes" configuration make sense for receive-encrypted folders?

Reply:
In my setup I would disable watching and set the rescan interval to 86400. Maybe it makes sense to offer that here as well?

Reply:
I'll not disable it entire, but use those as defaults.

Reply:
That means the UI also needs to know the "default default" (without user configuration), depending on some folder parameters. Now there's "default defaults" depending on recv-enc, but already before there was e.g. the rescan interval "default default" depending on watching being enabled. That's doable but I am at a loss regarding naming right now: "DefaultFolder` or `/rest/config/defaults/folder` is the name for user-configured defaults. What would be a good name to get a "default default"?    
`/rest/config/defaults/folder/default` feels disingenuous and obtuse. `/rest/config/defaults/unconfigured/folder` isn't quite nice either. Ideas?

Reply:
Holy shit. Just hardcode something? :D

Reply:
@imsodin That wasn't easy to follow, but I think I see what you're trying to achieve.  My thought was simpler:  When we have (configurable or even constant) defaults available from an API endpoint, that request gets an optional parameter for the folder type.  If given, the returned defaults are adjusted with anything that would be overruled by the backend anyway because of the folder type.  So if e.g. `ignorePerms` is a must (or strongly advised) for encrypted folders, then the GUI does not need to handle it specially, but just preselect the choice as given in the API response.  Of course the user could still change it (unless we really hide the option for encrypted folders) but the backend would revert to the required choice anyway.

So no "default defaults" for the GUI to compare against, but only the user's configured defaults, with some precautions applied based on the given folder type hint.  Does that make sense?

Reply:
That's endless: You could also have a parameter for fs watcher enabled or not, as that also influences the scan interval. And there might be more. Unless there's an elegant solution to cover it all, I'd rather keep it how it is now (some values in the UI). And even if there's an elegant solution: Unless it's super simple/small - not in this PR.

Reply:
> Unless it's super simple/small - not in this PR.

Definitely agree.